File: src/main/java/com/fyp/studentinterfaceservice/services/interfaces/ResumeService.java
package com.fyp.studentinterfaceservice.services.interfaces;

import com.fyp.studentinterfaceservice.model.Resume;
import com.fyp.studentinterfaceservice.model.User;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.ResponseEntity;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface ResumeService {

    Resume generateDCv(User user, String companyName, String jobTitle);

    ResponseEntity<InputStreamResource> getCv(String username);

    Resume uploadCv(MultipartFile file, Long userId) throws IOException;

    List<Resume> getAllCvs(Long userId);

    Resume updateCv(Resume resume);
}

File: src/main/resources/student-interface-service/src/app/service/resume.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {UserModel} from "../model/user.model";
import {ResumeModel} from "../model/resume.model";

@Injectable({
  providedIn: 'root'
})
export class ResumeService {

  constructor(private http: HttpClient) {
  }

  generateDynamicCv(user: UserModel, companyName: string, jobTitle: string): any {
    return this.http.post('http://localhost:8083/generateDynamicCv/' + companyName + '/' + jobTitle, user);
  }

  getCv(username: string): Observable<any> {
    return this.http.get('http://localhost:8083/getCv/' + username, {
      responseType: 'blob'
    });
  }

  getUserCvs(userId: number): Observable<any> {
    return this.http.get('http://localhost:8083/getAllCvs/' + userId);
  }

  uploadCv(cvData: FormData, userId: number): Observable<any> {
    return this.http.post('http://localhost:8083/upload/cv/' + userId,
      cvData);
  }

  updateCv(cv: ResumeModel): Observable<any> {
    return this.http.put('http://localhost:8083/updateCv/',
      {
        "resumeId": cv.resumeId,
        "name": cv.name,
        "data": cv.data,
        "studentId": cv.studentId
      });
  }

}
